Template/azureMonitorAlerts.json

```diff
@@ -0,0 +1,194 @@
+            "location": "global",
+            "properties": {
+                "description": "High error count (correctness)",
+                "severity": 2,
+                "enabled": true,
+                "scopes": [
+                    "[variables('ai_scope_resource_id')]"
+                ],
+                "evaluationFrequency": "[variables('evaluationFrequency')]",
+                "windowSize": "[variables('aggregationPeriod')]",
+                "criteria": {
+                    "allOf": [
+                        {
+                            "alertSensitivity": "[variables('alertSensitivity')]",
+                            "failingPeriods": {
+                                "numberOfEvaluationPeriods": "[variables('evaluationPeriods')]",
+                                "minFailingPeriodsToAlert": "[variables('failingPeriods')]"
+                            },
+                            "name": "Metric1",
+                            "metricNamespace": "[variables('metricNamespace')]",
+                            "metricName": "UnhandledExceptions",
+                            "operator": "GreaterThan",
+                            "timeAggregation": "Count",
+                            "criterionType": "DynamicThresholdCriterion",
+                            "skipMetricValidation": true
+                        }
+                    ],
+                    "odata.type": "Microsoft.Azure.Monitor.MultipleResourceMultipleMetricCriteria"
+                },
+                "autoMitigate": true,
+                "targetResourceType": "microsoft.insights/components",
+                "actions": []
+            }
+        },
+        {
+            "type": "microsoft.insights/metricAlerts",
+            "apiVersion": "2018-03-01",
+            "name": "High Downstream Messages Lost Ratio",
+            "location": "global",
+            "properties": {
+                "description": "High device messages lost ratio (correctness, throughput)",
+                "severity": 2,
+                "enabled": true,
+                "scopes": [
+                    "[variables('iot_hub_scope_resource_id')]"
+                ],
+                "evaluationFrequency": "[variables('evaluationFrequency')]",
+                "windowSize": "[variables('aggregationPeriod')]",
+                "criteria": {
+                    "allOf": [
+                        {
+                            "alertSensitivity": "[variables('alertSensitivity')]",
+                            "failingPeriods": {
+                                "numberOfEvaluationPeriods": "[variables('evaluationPeriods')]",
+                                "minFailingPeriodsToAlert": "[variables('failingPeriods')]"
+                            },
+                            "name": "Metric1",
+                            "metricNamespace": "Microsoft.Devices/IotHubs",
+                            "metricName": "c2d.commands.egress.abandon.success",
```

Comment:
Even though it's not in the ADR, should we maybe consider rejected messages here as well?

Reply:
That's a good point. I'll add a new alert.

Reply:
Added an alert based on c2d.commands.egress.reject.success